pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -179,7 +180,8 @@ func convertServices(cfg config.Config) []*model.Service {
 	if serviceEntry.WorkloadSelector != nil {
 		labelSelectors = serviceEntry.WorkloadSelector.Labels
 	}
-	hostAddresses := []*HostAddress{}
+
+	hostAddresses := make([]*HostAddress, 0, len(serviceEntry.Hosts)*slices.NonZeroLenOr(serviceEntry.Addresses, 1))
```

Comment:
we donot need such complexity, unless this is the perf bottle neck

Reply:
reverted